src/api/apiTypes.js

```diff
@@ -1,11 +1,185 @@
+ * these.
+ */
+export type SlimEventReaction = {
+  emoji_name: string,
+  user: any,
+};
+
+export type MessageEdit = {
+  prev_content?: string,
+  prev_rendered_content?: string,
+  prev_rendered_content_version?: number,
+  prev_subject?: string,
+  timestamp: number,
+  user_id: number,
+};
+
+export type Message = {
+  avatar_url: ?string,
+  client: string,
+  content: string,
+  content_type: 'text/html' | 'text/markdown',
+  display_recipient: any, // string | Recipient[],
+  edit_history: MessageEdit[],
+  flags: string[],
+  gravatar_hash: string,
+  id: number,
+  isOutbox: boolean,
+  is_me_message: boolean,
+  match_content?: string,
+  match_subject?: string,
+  reactions: SlimEventReaction[],
+  recipient_id: number,
+  sender_domain: string,
+  sender_email: string,
+  sender_full_name: string,
+  sender_id: number,
+  sender_realm_str: string,
+  sender_short_name: string,
+  stream_id: number,
+  subject: string,
+  subject_links: string[],
+  submessages: Message[],
+  timestamp: number,
+  type: 'stream' | 'private',
+};
+
+export type NarrowOperator =
+  | 'is'
+  | 'in'
+  | 'near'
+  | 'id'
+  | 'stream'
+  | 'topic'
+  | 'sender'
+  | 'pm-with'
+  | 'search';
+
+export type NarrowElement = {
+  operand: string,
+  operator?: NarrowOperator, // TODO type: this shouldn't be absent.
+};
+
+export type Narrow = NarrowElement[];
+
+export type RealmEmojiType = {
+  author: {
+    email: string,
+    full_name: string,
+    id: number,
+  },
+  deactivated: boolean,
+  id: number,
+  name: string,
+  source_url: string,
+  // This prevents accidentally using this type as a map.
+  // See https://github.com/facebook/flow/issues/4257#issuecomment-321951793
+  [any]: mixed,
+};
+
+export type RealmEmojiState = {
+  [id: string]: RealmEmojiType,
+};
+
+export type RealmFilter = [string, string, number];
+
+export type Stream = {
+  stream_id: number,
+  description: string,
+  name: string,
+  invite_only: boolean,
+  is_announcement_only: boolean,
+};
+
+export type Subscription = Stream & {
+  color: string,
+  in_home_view: boolean,
+  pin_to_top: boolean,
+  audible_notifications: boolean,
+  desktop_notifications: boolean,
+  email_address: string,
+  push_notifications: boolean,
+  is_old_stream: boolean,
+  push_notifications: boolean,
+  stream_weekly_traffic: number,
+};
+
+export type Topic = {
+  name: string,
+  max_id: number,
+};
+
+/**
+ * A Zulip user.
+ *
+ * For details on the properties, see the Zulip API docs:
+ *   https://zulipchat.com/api/get-all-users#response
+ */
+export type User = {
+  avatar_url: string,
+  bot_type?: number,
+  bot_owner?: string,
+  email: string,
+  full_name: string,
+  is_admin: boolean,
+  is_bot: boolean,
+  profile_data?: Object,
+  timezone: string,
+  user_id: number,
+};
+
 export type ApiResponse = {
   result: string,
   msg: string,
 };
 
+export type ApiResponseSuccess = {
+  result: 'success',
+  msg: '',
+};
+
+export type ApiErrorCode =
```

Comment:
This list is from the `ErrorCode` enum in the server's `zerver/lib/exceptions.py`, I think.

It'd be good to mention that in a short jsdoc comment.